analytic_engine/src/instance/write.rs

```diff
@@ -422,15 +422,23 @@ impl<'a> Writer<'a> {
 
         self.preprocess_write(&mut encode_ctx).await?;
 
-        let encoded_payload = {
-            let _timer = self.table_data.metrics.start_table_write_encode_timer();
-            let schema = self.table_data.schema();
-            encode_ctx.encode(&self.instance.wal_encode, &schema)?
-        };
+        let seq = if self.instance.disable_wal {
+            MIN_SEQUENCE_NUMBER
```

Comment:
The real sequence number should be returned, otherwise the merge procedure may not work with the mock one.